packages/@react-aria/searchfield/src/useSearchField.ts

```diff
@@ -73,7 +73,9 @@ export function useSearchField(
     }
 
     if (key === 'Escape') {
-      if (state.value === '') {
+      // Also check the inputRef value for the case where the value was set directly on the input element instead of going through
+      // the hook
+      if (state.value === '' && (!inputRef.current || inputRef.current.value === '')) {
```

Comment:
As mentioned above, currently there can be a situation where the value of the input field isn't passed in via props but is instead applied directly onto the input element and thus `state.value` here could be `''` but the value applied on the input is `"blah"` or something and thus we'd erroneously would continuePropagation. Check the ref's value as a fallback